guestbook/commentapp/tests.py

```diff
@@ -0,0 +1,101 @@
+from django.test import TestCase
+from datetime import timedelta
+
+from .models import GuestPost
+
+
+class TestViews(TestCase):
+
+    name = 'Guest Name'
+    comment = 'This is awesome!'
+    valid_email = 'guest@geocities.com'
+    invalid_email = 'ggg..$%$#sdlfkj%%'
+
+    def test_createWithoutEmail(self):
+        data = {
+            'name': self.name,
+            'comment': self.comment
+        }
+        response = self.client.post('/', data, follow=True)
+        self.assertRedirects(response, '/thankyou/', 302, 200)
+        response = self.client.get('/')
+        self.assertContains(response, data['name'])
+        self.assertContains(response, data['comment'])
+        self.assertContains(response, "published Just now")
+
+    def test_createWithValidEmail(self):
+        data = {
+            'name': self.name,
+            'email': self.valid_email,
+            'comment': self.comment
+        }
+        response = self.client.post('/', data, follow=True)
+        self.assertRedirects(response, '/thankyou/', 302, 200)
+        response = self.client.get('/')
+        self.assertContains(response, data['name'])
+        self.assertContains(response, data['comment'])
+        self.assertContains(response, "published Just now")
+
+    def test_createWithInvalidEmail(self):
+        data = {
+            'name': self.name,
+            'email': self.invalid_email,
+            'comment': self.comment
+        }
+        response = self.client.post('/', data, follow=True)
+        self.assertContains(response, "Enter a valid email address.")
+
+    def test_postCreatedWithDate(self):
+        data = {
+            'name': self.name,
+            'email': self.valid_email,
+            'comment': self.comment
+        }
+        response = self.client.post('/', data, follow=True)
+        self.assertRedirects(response, '/thankyou/', 302, 200)
+        recent_post = GuestPost.objects.get(pk=1)
```

Comment:
This might not always work, depending on your database the id might just continue counting up. You could use something like `recent_post = GuestPost.objects.last()`